### ISSUE TITLE
Add gradient clipping for autoencoder

### ISSUE THREAD

**Issue by nateraw** [Stability-AI/stable-audio-tools#110]
It seems you can't do automatic gradient clipping for autoencoder training, as we're using manual optimization in Lightning. Have seen this in the docs for adding it in manual optimization case: https://lightning.ai/docs/pytorch/stable/common/optimization.html#configure-gradient-clipping.

I believe by doing it this way the gradients will be unscaled properly when using mixed precision, but I'm not 100% sure.
